sources/packages/backend/apps/queue-consumers/src/services/application/application.service.ts

```diff
@@ -153,4 +153,67 @@ export class ApplicationService {
         .getMany()
     );
   }
+
+  /**
+   * Retrieves applications eligible for a specific notification (likely related to disability status and PD/PPD status).
+   * This method applies several criteria to filter eligible applications:
+   * - Study end date is at least 8 weeks in the future
+   * - Student's disability status is not 'PD' or 'PPD'
+   * - The assessment's workflow data indicates a positive PD/PPD status
+   * - The application is not archived
+   * - There's a pending disbursement schedule
+   * - No notification of this type (messageId 30) has been sent for this assessment
+   * @returns An array of eligible applications with relevant details for notification.
+   */
+  async getEligibleApplicationsForNotification(): Promise<Application[]> {
+    const eightWeeksFromNow = new Date();
+    eightWeeksFromNow.setDate(eightWeeksFromNow.getDate() + 56); // 8 weeks * 7 days
+
+    return this.applicationRepo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationNumber",
+        "users.firstName",
+        "users.lastName",
+        "users.email",
+        "users.id",
+        "student_assessments.id",
+        "students.disabilityStatus",
+      ])
+      .innerJoin("application.currentAssessment", "student_assessments")
+      .innerJoin("application.student", "students")
+      .innerJoin("students.user", "users")
+      .innerJoin("student_assessments.offering", "epo")
+      .innerJoin("student_assessments.disbursementSchedules", "ds")
+      .where("epo.studyEndDate >= :eightWeeksFromNow", { eightWeeksFromNow })
+      .andWhere("students.disabilityStatus NOT IN ('PD', 'PPD')")
+      .andWhere(
```

Comment:
We should use the JSONB operators as much as possible.
See the below example:
```SQL
select * from sims.student_assessments as
where workflow_data @> '{ "calculatedData": { "pdppdStatus": true}}'
```

Reply:
Please check the [shared query](https://github.com/bcgov/SIMS/pull/3759/files#r1788340560) with the implemented example.